sensors_control.cpp

```diff
@@ -105,73 +109,108 @@ void sensors_control_setup() {
+    a3_lit = gSensorA3_light;
+  }
+  ATOMIC_BLOCK(ATOMIC_RESTORESTATE) { 
+    a4_dark = gSensorA4_dark;
+    a4_lit = gSensorA4_light;
+  }
+  ATOMIC_BLOCK(ATOMIC_RESTORESTATE) { 
+    a5_dark = gSensorA5_dark;
+    a5_lit = gSensorA5_light;
+  }
+  
+  if(mode == 'd') // the default
   {
-    Serial.print(max(gSensorA0_light_ - gSensorA0_dark_, 0));
+    Serial.print(max(a0_lit - a0_dark, 0));
+    Serial.print(comma);
+    Serial.print(max(a1_lit - a1_dark, 0));
     Serial.print(comma);
-    Serial.print(max(gSensorA1_light_ - gSensorA1_dark_, 0));
+    Serial.print(max(a2_lit - a2_dark, 0));
     Serial.print(comma);
-    Serial.print(max(gSensorA2_light_ - gSensorA2_dark_, 0));
+    Serial.print(max(a3_lit - a3_dark, 0));
     Serial.print(comma);
-    Serial.print(max(gSensorA3_light_ - gSensorA3_dark_, 0));
+    Serial.print(max(a4_lit - a4_dark, 0));
+    Serial.print(comma);
+    Serial.print(max(a5_lit - a5_dark, 0));
   }
-  else if(mode == 'h')
+  else if(mode == 'h') // display as hex values
   {
-    gSensorA0_light_ = constrain(gSensorA0_light_-gSensorA0_dark_, 0, 255);
-    if(gSensorA0_light_ < 0x10) { Serial.print('0'); }
-    Serial.print(gSensorA0_light_, HEX);
-    gSensorA1_light_ = constrain(gSensorA1_light_-gSensorA1_dark_, 0, 255);
-    if(gSensorA1_light_ < 0x10) { Serial.print('0'); }
-    Serial.print(gSensorA1_light_, HEX);
-    gSensorA2_light_ = constrain(gSensorA2_light_-gSensorA2_dark_, 0, 255);
-    if(gSensorA2_light_ < 0x10) { Serial.print('0'); }
-    Serial.print(gSensorA2_light_, HEX);
-    gSensorA3_light_ = constrain(gSensorA3_light_-gSensorA3_dark_, 0, 255);
-    if(gSensorA3_light_ < 0x10) { Serial.print('0'); }
-    Serial.print(gSensorA3_light_, HEX);
+    uint8_t diff;
+    print_hex2(a0_lit-a0_dark);
+    print_hex2(a1_lit-a1_dark);
+    print_hex2(a2_lit-a2_dark);
+    print_hex2(a3_lit-a3_dark);
+    print_hex2(a4_lit-a4_dark);
+    print_hex2(a5_lit-a5_dark);
   }
   else if(mode == 'r')
   {
-    Serial.print(gSensorA0_dark_);
+    Serial.print(a0_dark);
+    Serial.print(comma);
+    Serial.print(a1_dark);
+    Serial.print(comma);
+    Serial.print(a2_dark);
     Serial.print(comma);
-    Serial.print(gSensorA1_dark_);
+    Serial.print(a3_dark);
     Serial.print(comma);
-    Serial.print(gSensorA2_dark_);
+    Serial.print(a4_dark);
     Serial.print(comma);
-    Serial.print(gSensorA3_dark_);
+    Serial.print(a5_dark);
     Serial.print(comma);
-    Serial.print(gSensorA0_light_);
+    Serial.print(' ');
+    Serial.print(' ');
+    Serial.print(a0_lit);
     Serial.print(comma);
-    Serial.print(gSensorA1_light_);
+    Serial.print(a1_lit);
     Serial.print(comma);
-    Serial.print(gSensorA2_light_);
+    Serial.print(a2_lit);
     Serial.print(comma);
-    Serial.print(gSensorA3_light_);
+    Serial.print(a3_lit);
+    Serial.print(comma);
+    Serial.print(a4_lit);
+    Serial.print(comma);
+    Serial.print(a5_lit);
   }
   Serial.println();
+  delay(100);
```

Comment:
Why is there a delay 100? Left over from testing?

Reply:
Yes, my mistake. There is also the looped nature of the response to take out. I will do that later today.